### PR TITLE
add logs to relay chain selection

### PR DESCRIPTION
**Pull request by drahnr**
For debugability, adds relay chain selection logs.

### CODE REVIEW

File: node/service/src/relay_chain_selection.rs [hunk 1]
```diff
@@ -409,6 +426,12 @@ where
 			}
 		};
 
+		tracing::trace!(
+			target: LOG_TARGET,
+			"Ancestor approval restriction applied: {:?}",
+			&subchain_head
```
Comment (bkchr) [inline node/service/src/relay_chain_selection.rs:432]:
```suggestion
			?subchain_head,
			"Ancestor approval restriction applied",
```

The nice thing about the tracing log macros is that you can forward this information much better than putting them into "text".

**Review by ordian — APPROVED** [review]
Please add logs for `best_chain` and then it's good to go.

**Comment by drahnr** [comment]
> Please add logs for `best_chain` and then it's good to go.

https://github.com/paritytech/polkadot/pull/3990/files#diff-cb277cc4cd542f9f6daab01f6c7894b5ec8131e1f226e10a5524a9fcb5e05780R303-R316

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**